tests/integration/server/nonclassevents/nonclassevent.controller.test.ts

```diff
@@ -107,4 +127,60 @@ describe('Non Class Event API', function () {
       });
     });
   });
+  describe('POST /', function () {
+    describe('User is not authenticated', function () {
+      it('is inaccessible to unauthenticated users', async function () {
+        authStub.rejects(new ForbiddenException());
+
+        const response = await request(api)
+          .post('/api/non-class-events')
+          .send(createNonClassParent);
+
+        strictEqual(response.ok, false);
+        strictEqual(response.status, HttpStatus.FORBIDDEN);
+        strictEqual(
+          mockNonClassEventService.createWithNonClassEvents.called,
+          false
+        );
+      });
+    });
+    describe('User is authenticated', function () {
+      describe('User is not authorized', function () {
+        beforeEach(function () {
+          authStub.rejects(new UnauthorizedException());
+        });
+        it('is inaccessible to unauthorized users', async function () {
+          const response = await request(api)
+            .post('/api/non-class-events')
+            .send(createNonClassParent);
+
+          strictEqual(response.ok, false);
+          strictEqual(response.status, HttpStatus.UNAUTHORIZED);
+          strictEqual(
+            mockNonClassEventService.createWithNonClassEvents.called,
+            false
+          );
+        });
+      });
+      describe('User is authorized', function () {
+        beforeEach(function () {
+          authStub.resolves(nonClassEventManager);
+        });
+        it('creates non-class parent data', async function () {
+          mockAreaRepository.findOneOrFail.resolves(rawAreaList[0]);
```

Comment:
I don't think `rawAreaList` is the right data to use here, as it only returns a `name` field instead of a full `Area` entity.

Additionally, it seems that none of these tests are actually testing that `areaRepository.findOneOrFail` is actually searching correctly with the data in the request -- Anywhere the controller route is invoked the method is just being stubbed out to return data. I think that's created a blind spot in our tests, as it's not picking up that the controller code is searching by name, while the DTO is enforcing that `area` should be the id of an area. One solution would be to combine these `service` and `controller` integration tests, and test everything against the database using only the API method. (That's what I've taken to doing, e.g. the [Instructor PUT method][1])

[1]: https://github.com/seas-computing/course-planner/blob/develop/tests/integration/server/courseInstance/courseInstance.controller.test.ts#L584

Reply:
Just wanted to clarify this comment from Friday: I'm not recommending rewriting all the tests to run everything through the API, just highlighting it as a strategy for future tests.

In this case I think we just need to:

- [ ] Return a full `Area` entity
- [ ] Correct the shape of the data to match the DTO (or vice versa)

Reply:
I've updated the repository to now resolve to an instance of `Area` and corrected the way the controller behaves to treat the incoming `area` value as a UUID instead of the area name. 

Both of these were addressed in 86ca9acddef9a66a15a0f8b33583a68af058f18d